gap/tools/io.gi

```diff
@@ -313,3 +313,212 @@ function(str)
 
   return IteratorByFunctions(record);
 end);
+
+InstallGlobalFunction(ReadMultiplicationTable,
+function(arg)
+  local name, line_nr, file, ReadMultiplicationTableLine, i, line, n, lines;
+  if Length(arg) = 1 then
+    name    := arg[1];
+    line_nr := 0;
+  elif Length(arg) = 2 then
+    name    := arg[1];
+    line_nr := arg[2];
+  else
+    ErrorNoReturn("Semigroups: ReadMultiplicationTable: usage,\n",
+                  "there should be at most 2 arguments,");
+  fi;
+  if IsString(name) then
+    name := UserHomeExpand(name);
+    file := IO_CompressedFile(name, "r");
+    if file = fail then
+      ErrorNoReturn("Semigroups: ReadMultiplicationTable:\n",
+                    "could not open the file \"", name, "\",");
+    fi;
+  elif IsFile(name) then
+    file := name;
+  else
+    ErrorNoReturn("Semigroups: ReadMultiplicationTable: usage,\n",
+                  "the first argument must be a string or a file,");
+  fi;
+  if not (IsInt(line_nr) and line_nr >= 0) then
+    ErrorNoReturn("Semigroups: ReadMultiplicationTable: usage,\n",
+                  "the second argument must be a positive integer,");
+  fi;
+  ReadMultiplicationTableLine := SEMIGROUPS.ReadMultiplicationTableLine;
+
+  if line_nr <> 0 then
+    i := 0;
+    repeat
+      i    := i + 1;
+      line := IO_ReadLine(file);
+    until i = line_nr or line = "";
+    if IsString(arg[1]) then
+      IO_Close(file);
+    elif line = "" then
+      ErrorNoReturn("Semigroups: ReadMultiplicationTable:\n",
+                    "the file only has ", i - 1, " lines,");
+    fi;
+    n := RootInt(Length(line)); #TODO: Returns the int part. Change or check?
```

Comment:
Maybe just add some checks here that length of the line is indeed a perfect square